modules/core/include/visp3/core/vpImageTools.h

```diff
@@ -796,6 +799,28 @@ void vpImageTools::undistort(const vpImage<Type> &I, const vpCameraParameters &c
 #endif
 }
 
+/*!
+  Undistort an image distorted by fisheye lens
+
+  \param I : Input image to undistort.
+
+  \param cam : Parameters of the camera causing distortion.
+
+  \param undistI : Undistorted output image. The size of this image
+  will be the same than the input image \e I.
+
+  \sa initUndistortMap, remap
+*/
+template<class Type>
+void vpImageTools::undistortFisheye(const vpImage<Type> &I, const vpCameraParameters &cam, vpImage<Type> &undistI)
+{
+  vpArray2D<int> mapU, mapV;
+  vpArray2D<float> mapDu, mapDv;
+
+  initUndistortMap(cam, I.getWidth(), I.getHeight(), mapU, mapV, mapDu, mapDv);
```

Comment:
This will do each time `initUndistortMap` + `remap` whereas `initUndistortMap` can be done once for a particular distortion coefficients. This should matter when processing a camera stream.

A solution I see is to update `undistort()` code to remove the old pthread implementation and use instead the combination `initUndistortMap` + `remap`.

To be checked but I don't think the performance between the original `undistort()` and `initUndistortMap` + `remap` should be significant. Otherwise, if it is too much work to adapt `undistort()` with the pthread/non pthread code to handle the two camera models, it should be fine to keep `undistortFisheye()`.

Reply:
As you mention, the plan is to call  `initUndistortMap()` + `remap()` in `undistort()` if performances are similar with the old implementation and to add a test to call `initUndistortMap()` only when camera parameter change or when image size change.

Reply:
Just a suggestion to use function overloading:
  - one `undistort()` function with `vpCameraParameters` as parameters to do `initUndistortMap()` + `remap()`
  - another `undistort()` function with the maps as parameters to just do `remap()` inside

Since `undistort()` is static, probably the solution is to have the image size and the camera parameters variables static inside the `undistort()` function.
I am not fan of using static variables for the following reasons:
  - what if we want to call `undistort()` from different threads (possible data race issue)?
  - what if we want to undistort simultaneously using two camera parameters a sequence of images?

---

Other than that, this is a cool new feature :+1:

Reply:
Yesterday, I did the changes related to `vpRealSense2` and `vpCameraParameters` classes. I will be fixing today `undistort()` according to your suggestions.